### ISSUE TITLE
MKS TFT 28 doesn't connect to RAMPS

### ISSUE THREAD

**Issue by martind-n** [makerbase-mks/MKS-TFT#60]
Hi,

I'm facing an issue with mi MKS TFT 28. I'm trying to attach it to a RAMPS 1.4 board, however there is no communication at all. 
I tried to change baud rate on both boards but nothing. 
I'm also supplying 12V from an external PS (the same that powers RAMPS).
Funny thing is that running some USB connection (like pronterface) and being TFT28 connected, the screen shows the the temperature accordingly. I know I should use USB connected at the same time running TFT28, but it was just another test I made...

Is there an other thing to look at in order to get the board working?

**Comment by makerbase-mks** [comment]
Is the serial line reverse?

**Comment by martind-n** [comment]
No... On the MKS board there is only one position. On RAMPS I'm putin the red line pointing to power supply connector.

**Comment by Ganriber** [comment]
I'm sorry, you probably wrote this letter to me by mistake and it is waiting for someone else. -- С уважением Макаров Павел Владимирович...   25.10.2017, 05:39, "martind-n" <notifications@github.com>:No... On the MKS board there is only one position. On RAMPS I'm putin the red line pointing to power supply connector.—You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by OskarLM** [comment]
I have the same problem. any solution?

**Comment by martind-n** [comment]
Oskar,
I could solve my issue by changing the serial port on my Arduino (using serial2 instead of 0). That makes the screen communicates to printer, however there's another problem regarding printing from SD card.
You can take a look to this thread, still no answer from MKS (In fact, I open my own thread about this and no reply either)
https://github.com/makerbase-mks/MKS-TFT/issues/8
So, my workaround will work only if you don't face the SD printing issue if not it won't fit for you, because changing the default serial port will prevent other tools (such as pronter face) communicate with Marlin.

**Comment by dudisr** [comment]
i have the same problem, on ramps red cable same direction to power connector but i still cannot control by MKS

**Comment by Devrim78** [comment]
Aynı  sorunu bende yaşıyorum.. Bir ürün sadece sattık ve oldu demekle olmaz maalesef, bir ürün satıldığı zaman arkasında durulması lazım. Ramps1.4 ile uyumlu yazılıyor fakat ne yaptıysak ürün çalışmıyor. Yazılan bütün her şeyi yaptım. Lütfen detaylı bir açıklamaya yaparsanız sevinirim....

**Comment by Kodbee** [comment]
I have the same problem: MKS TFT28 V4 don't communicate with Ramps 1.4. any solution?

**Comment by dudisr** [comment]
Even though is not a solution, i changed the ramps for skr1.3 the communication with board works flawlessly

**Comment by makerbase-mks** [comment]
1. MKS TFT28 use DC12V power by DC terminal
2. Set the motherboard baud rate and connect AUX-1 correctly

**Comment by Kodbee** [comment]
Thanks. I did, but still not connecting. This does not work in any way

**Comment by martind-n** [comment]
Hi. What I actually did at that time was to change default por in Marlin for port #2 and then wired TFT TX to ramps pin #17 (RX) and TFT RX to ramps pin #16 (TX). By doing this I could get my tft and ramps working, however, the side effect is pronterface stop working, so mind your decision before doing this!

**Comment by Kodbee** [comment]
Thanks Martind-n, but we need that it works fine with both ;) ... Does anyone solve this problem between MKS TFT28 V4.0 and Ramps?

**Comment by makerbase-mks** [comment]
Should be caused by RAMPS problems. Need to see how the corresponding pins are distributed

**Comment by Kodbee** [comment]
Thanks, I send you the pictures of how the the corresponding pins are distributed... It´s impossible to connect...
![IMG_20200606_122944](https://user-images.githubusercontent.com/48182942/83942269-6c5a4180-a7f2-11ea-8631-6111647a2236.jpg)
![IMG_20200606_122531](https://user-images.githubusercontent.com/48182942/83942271-6e240500-a7f2-11ea-9b02-d1b5acc95f42.jpg)
![IMG_20200606_122421](https://user-images.githubusercontent.com/48182942/83942282-7a0fc700-a7f2-11ea-871f-be0826815c44.jpg)

**Comment by makerbase-mks** [comment]
MKS TFT28 AUX-1 pin
![image](https://user-images.githubusercontent.com/12979070/84132562-9cb31180-aa78-11ea-9f99-3398966ad31a.png)

**Comment by Kodbee** [comment]
The pins are correct. I have tried another cable, following the indications in the diagram that you have sent me, and it still does not connect.
![IMG_20200609_120605](https://user-images.githubusercontent.com/48182942/84135530-6363ad00-aa4a-11ea-8b42-31f0205ff90a.jpg)

**Comment by Kodbee** [comment]
It´s impossible. MKS TFT28 4.0 **DOESN´T WORK** with Ramps 1.4. I am trying communicate both since 3 months ago, but it is impossible. MKS´s people don´t solve this issue. Money waste...

**Comment by daniellezama** [comment]
Hi from argentina, I have the same problem Using Marlin 2.0 bugfix with a MKS SGENL V1.0 and a MKS TFT 3.5 with wifi. All in 250000 Bauds and no comunication. The display is on the wifi is conected, is in marlin mode, but no works the functions when i try to use the tft. With the reprap full graphics all works fine. Any have an idea???

Regards

Daniel

**Comment by makerbase-mks** [comment]
@daniellezama
You can refer to MKS SGEN_L Wiki: https://github.com/makerbase-mks/SGEN_L/wiki/Display_MKS_TFT35

**Comment by daniellezama** [comment]
> @daniellezama
> You can refer to MKS SGEN_L Wiki: https://github.com/makerbase-mks/SGEN_L/wiki/Display_MKS_TFT35

Yes I made this and works!! this is not commented in all tutorials that i have seen before. 

Regards

Printblaster
